comp/languagedetection/client/client.go

```diff
@@ -153,28 +159,35 @@ func (c *client) run() {
 	)
 	defer c.store.Unsubscribe(eventCh)
 
-	metricTicker := time.NewTicker(runningMetricPeriod)
-	defer metricTicker.Stop()
-
-	retryProcessWithoutPodTicker := time.NewTicker(c.retryProcessWithoutPodPeriod)
-	defer retryProcessWithoutPodTicker.Stop()
+	cleanupProcessesWithoutPodCleanupTicker := time.NewTicker(c.processesWithoutPodCleanupPeriod)
+	defer cleanupProcessesWithoutPodCleanupTicker.Stop()
 
 	go c.startStreaming()
 
 	for {
 		select {
 		case eventBundle := <-eventCh:
-			c.processEvent(eventBundle)
-		case <-retryProcessWithoutPodTicker.C:
-			c.retryProcessEventsWithoutPod()
+			c.handleEvent(eventBundle)
+		case <-cleanupProcessesWithoutPodCleanupTicker.C:
+			c.cleanUpProcesssesWithoutPod(time.Now())
 		case <-c.ctx.Done():
 			return
 		}
 	}
 }
 
-// processEvent processes events from workloadmeta
-func (c *client) processEvent(evBundle workloadmeta.EventBundle) {
+func (c *client) cleanUpProcesssesWithoutPod(now time.Time) {
```

Comment:
Part of the new retry logic if we detect a process without detecting its corresponding pod. There is a race between the kubelet collector and the process collector.